src/main/routes/metrics-route.ts

```
@@ -22,11 +23,8 @@ async function loadMetrics(promQueries: string[], cluster: Cluster, prometheusPa
         try {
           return await cluster.getMetrics(prometheusPath, { query, ...queryParams })
         } catch (error) {
-          if (lastAttempt || error?.statusCode === 404) {
-            return {
-              status: error.toString(),
-              data: { result: [] },
-            }
+          if (lastAttempt || (error?.statusCode >= 400 && error?.statusCode < 500)) {
```

Fail fast if getting any error response